app/server/src/features/accounts/methods.unit.test.ts

```diff
@@ -0,0 +1,463 @@
+
+        prismaMock.user.findUnique.mockResolvedValueOnce(expectedOutput);
+
+        const output = await AccountMethods.findUserById(userData.id, prismaMock);
+
+        expect(output).toEqual(expectedOutput);
+    });
+});
+
+/*
+ *  test cases for getting organizations of a particular user
+ */
+describe('findOrgsByUserId', () => {
+    test('should find a single org', async () => {
+        const user = {
+            ...userData,
+            memberOf: [
+                {
+                    organization: 'Prytaneum',
+                },
+            ],
+        };
+
+        prismaMock.user.findUnique.mockResolvedValueOnce(user);
+
+        const expectedOutput = ['Prytaneum'];
+
+        const output = await AccountMethods.findOrgsByUserId(userData.id, prismaMock);
+
+        expect(output).toEqual(expectedOutput);
+    });
+
+    test('should find multiple orgs', async () => {
+        const user = {
+            ...userData,
+            memberOf: [
+                {
+                    organization: 'Prytaneum',
+                },
+                {
+                    organization: 'University of California, Riverside',
+                },
+                {
+                    organization: 'School of Public Policy',
+                },
+                {
+                    organization: 'Southern California',
+                },
+            ],
+        };
+
+        prismaMock.user.findUnique.mockResolvedValueOnce(user);
+
+        const expectedOutput = [
+            'Prytaneum',
+            'University of California, Riverside',
+            'School of Public Policy',
+            'Southern California',
+        ];
+
+        const output = await AccountMethods.findOrgsByUserId(userData.id, prismaMock);
+
+        expect(output).toEqual(expectedOutput);
+    });
+
+    test('should not find orgs', async () => {
+        prismaMock.user.findUnique.mockRejectedValueOnce(null);
+
+        await expect(AccountMethods.findOrgsByUserId(userData.id, prismaMock)).rejects.toBeNull();
+    });
+});
+
+/*
+ *  test cases for the function called when a user is registering themselves
+ */
+describe('registerSelf', () => {
+    test('should register a user if passwords match', async () => {
+        const input = {
+            email: userData.email,
+            firstName: userData.firstName,
+            lastName: userData.lastName,
+            password: userTextPassword,
+            confirmPassword: userTextPassword,
+        };
+
+        // mock registered user
+        const registeredUser = { ...userData };
+
+        prismaMock.user.create.mockResolvedValueOnce(registeredUser);
+
+        const userWithGlobalId = toUserId(registeredUser);
+        const token = await jwt.sign({ id: userWithGlobalId.id });
+
+        const expectedOutput = { registeredUser, token };
+
+        const output = await AccountMethods.registerSelf(prismaMock, input);
+
+        expect(output).toEqual(expectedOutput);
```

Comment:
```ts
// arrange
const input = {
            email: userData.email,
            firstName: userData.firstName,
            lastName: userData.lastName,
            password: userTextPassword,
            confirmPassword: userTextPassword,
        };
const registeredUser = { ...userData };
prismaMock.user.create.mockResolvedValueOnce(registeredUser);

// act
const output = await AccountMethods.registerSelf(prismaMock, input);

// assert
const userWithGlobalId = toUserId(registeredUser);
const token = await jwt.sign({ id: userWithGlobalId.id });
const expectedOutput = { registeredUser, token };
expect(output).toEqual(expectedOutput);
```
NOTE: it's fine to put the expected output generation in the assert block.  It would normally be made inline. I just think saving to a variable is nicer personally.